geps/gep-1713/index.md

```diff
@@ -0,0 +1,408 @@
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: httproute-example
+spec:
+  parentRefs:
+  - name: some-workload-listeners
+    kind: ListenerSet
+    sectionName: foo
+```
+
+#### Listener Validation
+
+Implementations MUST treat the parent `Gateway` as having the concatenated list of all listeners from itself and attached `ListenerSets`
+Validation of this list of listeners MUST behave the same as if the list were part of a single `Gateway`.
+
+From the earlier example the above resources would be equivalent to a single `Gateway` where the listeners are collapsed into a single list.
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: Gateway
+metadata:
+  name: parent-gateway
+spec:
+  gatewayClassName: example
+  listeners:
+  - name: foo
+    hostname: foo.com
+    protocol: HTTP
+    port: 80
+  - name: first
+    hostname: first.foo.com
+    protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: first-workload-cert # Provisioned via HTTP01 challenge
+  - name: second
+    hostname: second.foo.com
+    protocol: HTTPS
+    port: 443
+    tls:
+      mode: Terminate
+      certificateRefs:
+      - kind: Secret
+        group: ""
+        name: second-workload-cert # Provisioned via HTTP01 challenge
+```
+
+#### Listener Precedence
+
+Listeners should be merged using the following precedence:
+1. "parent" Gateway
+2. ListenerSet ordered by creation time (oldest first)
+3. ListenerSet ordered alphabetically by “{namespace}/{name}”.
+
+If there are listener conflicts, this should be reported as `Conflicted=True` in the ListenerStatus as usual. See 'Conditions' section below for more details on object should report the conflict.
+
+###  Gateway Conditions
+
+`Gateway` currently supports the following top-level condition types: `Accepted` and `Programmed`
+
+For a `Gateway`, `Accepted` should be set based on the entire set of merged listeners.
+For instance, if a `ListenerSet` listener is invalid, `ListenersNotValid` would be reported. 
+`Programmed` is not expected, generally, to depend on the children resources, but if an implementation does depend on these
+they should consider child resources when reporting this status.
+
+Parent gateways MUST NOT have `ListenerSet` listeners in their `status.listeners` conditions list.
+
+It is up to the implementation whether an invalid listener affects other listeners in the Gateway.
+
+### ListenerSet Conditions
+
+`ListenerSets` MUST NOT have their parent `Gateway`'s' listeners in the `status.listeners` conditions list.  An implementation MAY reject listeners with `ListenerConditionAccepted=False` and Reason `TooManyListeners` `ListenerSets`, like a `Gateway`, also have two top-level conditions: `Accepted` and `Programmed`. These conditions, when surfacing details about listeners, MUST only summarize the `status.listener` conditions that are exclusive to the `ListenerSet`. 
```

Comment:
I think we need to start with the basic `Accepted` behavior, which should be the same as Routes:

The `Accepted` Condition must be set on every ListenerSet, and indicates that the ListenerSet is semantically valid and accepted by its `parentRef`. It should have similar Reasons to the Route `Accepted` condition, like `NotAllowedByGateway`, `NoMatchingParent`, `UnsupportedValue` or the like. Following on from below, we'll probably also need a `ParentNotAccepted` or something to cover the case that the referenced Gateway is correctly configured by itself has a problem.

ListenerSets should also reuse the `ResolvedRefs` Condition with the same meaning as on Routes.

`Programmed` should also be present and have a similar meaning to the Gateway `Programmed` Condition, but agreed that it should _only_ reflect the Listeners in this ListenerSet. (tbh this may be pretty tricky to implement, I suspect).

Reply:
I split the section and added the conditions that I think are appropriate.

eg. ResolvedRefs in my mind doesn't work here because the ListenerSet doesn't reference any backends